src/lsp/ext.rs

```diff
@@ -89,6 +89,24 @@ pub mod testing {
     }
 }
 
+/// Notifies about diagnostics generation which is beginning to calculate
+#[derive(Debug)]
+pub struct DiagnosticsCalculationStart;
+
+impl Notification for DiagnosticsCalculationStart {
+    type Params = ();
+    const METHOD: &'static str = "cairo/diagnosticsCalculationStart";
+}
+
+/// Notifies about diagnostics generation which ended calculating
+#[derive(Debug)]
+pub struct DiagnosticsCalculationFinish;
+
+impl Notification for DiagnosticsCalculationFinish {
+    type Params = ();
+    const METHOD: &'static str = "cairo/diagnosticsCalculationFinish";
+}
```

Comment:
we wanted to have a single "analysis status" notificaiton, didn't we?

Reply:
It's the concept in RA really 
Do you believe that it will be more future proof?

Reply:
I think that's what we wanted, to have an enum of states like `Idle`, `ExpandingProcMacros`, `Analysing`. This would be troublesome to implement if we start doing `x started`/`x finished` notifications.